src/main/kotlin/com/manimdsl/runtime/DataStructureShapes.kt

```diff
@@ -246,21 +270,23 @@ class Scene {
 
     private fun centralise(fullScreen: Boolean = false) {
         if (sceneShapes.isNotEmpty()) {
-            val leftXCoord = sceneShapes.map { it.corners()[0] }.minOf { it.first }
-            val rightXCoord = sceneShapes.map { it.corners()[1] }.maxOf { it.first }
-            val topYCoord = sceneShapes.map { it.corners()[0] }.maxOf { it.second }
-            val bottomYCoord = sceneShapes.map { it.corners()[2] }.minOf { it.second }
+            val centeringShapes = sceneShapes.filter { it.canCentralise }
+
+            val leftXCoord = centeringShapes.map { it.corners()[0] }.minOfOrNull { it.first }
```

Comment:
This is the only part I am unsure about - it made the tests pass but it might not be correct. Without using `maxOfOrNull` and `minOfOrNull` (instead of just `maxOf` and `minOf`), Kotlin was throwing a `NoSuchElementException`.

The default values below might not be ideal either

Reply:
Oh I think we shouldn't do it like this, if centeringShapes is empty just skip this whole block of code

Reply:
Should be done now